File: src/main/java/ar/com/phostech/exceptions/ErrorCatalog.java
package ar.com.phostech.exceptions;

import static java.net.HttpURLConnection.*;

public enum ErrorCatalog {


    UNKNOWN(HTTP_INTERNAL_ERROR, ErrorCodes.UNKNOWN),
    INTERNAL_SERVER_ERROR(HTTP_INTERNAL_ERROR, ErrorCodes.INTERNAL_ERROR),
    VALIDATION_FAILED(HTTP_BAD_REQUEST,ErrorCodes.VALIDATION_ERROR),
    OPERATION_FAILED(HTTP_BAD_REQUEST,ErrorCodes.OPERATION_FAILED),
    NOT_FOUND(HTTP_NOT_FOUND,ErrorCodes.NOT_FOUND)
    ;

    public final int status;
    public final String defaultMessage;
    public final String error;

    ErrorCatalog(int _status, ErrorCodes errorCode) {
        status = _status;
        defaultMessage = errorCode.message;
        error = errorCode.error;
    }

    public static ErrorCatalog from(String id) {
        for (ErrorCatalog anEnum : values()) {
            if (anEnum.name().equalsIgnoreCase(id)) {
                return anEnum;
            }
        }
        return null;
    }
}

File: minipoc/src/main/kotlin/ar/com/phostech/demo/exceptions/ErrorCatalog.kt
package ar.com.phostech.demo.exceptions

import ar.com.phostech.vertx.response.ErrorDetail
import java.io.Serializable
import java.net.HttpURLConnection

enum class ErrorCatalog
(
    val httpStatus: Int,
    val errorCode: String,
    val defaultMessage: String
) : Serializable {
    // 200 OK
    OPERATION_IN_PROGRESS(HttpURLConnection.HTTP_CREATED, "operation.running", ""),
    // 400 CLIENT FAILED
    REQUIRED_ARGUMENT(HttpURLConnection.HTTP_BAD_REQUEST, "required.field", "Field must not be null"),
    INVALID_STATUS(HttpURLConnection.HTTP_BAD_REQUEST, "invalid.status", "Invalid order status."),
    VALIDATION_FAILED(HttpURLConnection.HTTP_BAD_REQUEST, "validation.failed", "Validation Failed"),
    // 500 Failed
    UNKNOWN(HttpURLConnection.HTTP_INTERNAL_ERROR, "unknown.error", "Unknown error"),
    METHOD_NOT_IMPLEMENTED(HttpURLConnection.HTTP_NOT_IMPLEMENTED, "not.implemented", "Method not implemented"),
    ;

    fun asErrorDetail(): ErrorDetail {
        return ErrorDetail(
            errorCode,
            defaultMessage
        )
    }

}

